src/flownet/ahm/_ahm_iteration_analytics.py

```diff
@@ -271,35 +272,43 @@ def make_dataframe_simulation_data(path, eclbase_file, keys):
     # Prepare dataframe
     # pylint: disable-msg=too-many-locals
     df_sim = pd.DataFrame()
-    for id_real, runpath in enumerate(realizations_dict.keys()):
+    id_real = 0
+    for runpath in realizations_dict:
         df_tmp = pd.DataFrame()
-        dates = realizations_dict[runpath].dates
-        if id_real == 0:
-            df_sim["DATE"] = pd.Series(dates)
-            df_sim["REAL_ID"] = pd.Series(id_real * np.ones(len(dates)), dtype=int)
-        df_tmp["DATE"] = pd.Series(dates)
-        df_tmp["REAL_ID"] = pd.Series(id_real * np.ones(len(dates)), dtype=int)
-
-        if id_real == 0:
-            for counter, k in enumerate(keys):
-                if counter == 0:
-                    key_list_data = [
-                        x for x in realizations_dict[runpath] if x.startswith(k)
-                    ]
-                else:
-                    key_list_data.extend(
-                        [x for x in realizations_dict[runpath] if x.startswith(k)]
-                    )
+        if (realizations_dict[runpath] is not None) and (
+            np.datetime64(realizations_dict[runpath].dates[-1]) >= end_date
+        ):
+            dates = realizations_dict[runpath].dates
+            if id_real == 0:
+                df_sim["DATE"] = pd.Series(dates)
+                df_sim["REAL_ID"] = pd.Series(id_real * np.ones(len(dates)), dtype=int)
+            df_tmp["DATE"] = pd.Series(dates)
+            df_tmp["REAL_ID"] = pd.Series(id_real * np.ones(len(dates)), dtype=int)
 
-        for key in key_list_data:
             if id_real == 0:
-                df_sim[key] = pd.Series(realizations_dict[runpath].numpy_vector(key))
-            df_tmp[key] = pd.Series(realizations_dict[runpath].numpy_vector(key))
+                for counter, k in enumerate(keys):
+                    if counter == 0:
+                        key_list_data = [
+                            x for x in realizations_dict[runpath] if x.startswith(k)
+                        ]
+                    else:
+                        key_list_data.extend(
+                            [x for x in realizations_dict[runpath] if x.startswith(k)]
+                        )
+
+            for key in key_list_data:
+                if id_real == 0:
+                    df_sim[key] = pd.Series(
+                        realizations_dict[runpath].numpy_vector(key)
+                    )
+                df_tmp[key] = pd.Series(realizations_dict[runpath].numpy_vector(key))
```

Comment:
Not sure why you are using the tmp df at all?

```suggestion
            for key in key_list_data:
                df_sim[key] = pd.Series(
                    realizations_dict[runpath].numpy_vector(key)
                )

```

Reply:
I tried the suggested changes and the code no longer works. We need df_tmp because each realization might have different dates due to numerical solution of the simulations. So using the auxiliary df_tmp was the way that I found to be able to load all the data into the same dataframe